pkg/registries/registries.go

```diff
@@ -24,47 +65,52 @@ type mirrorSet struct {
 	mirrorSourcePolicy config.MirrorSourcePolicy
 }
 
-// registryConfig represents a system-wide image registry configuration
-type registryConfig struct {
-	sourceConfigs []mirrorSet
+// newMirror constructs an object with proper source and mirror name structures to be used in containerd registry config
+func newMirrorSet(srcImage string, mirrorLocations []config.ImageMirror, resolveTags bool,
+	mirrorSourcePolicy config.MirrorSourcePolicy) mirrorSet {
+	truncatedMirrors := []mirror{}
+	for _, m := range mirrorLocations {
+		truncatedMirrors = append(truncatedMirrors, newMirror(srcImage, string(m), resolveTags))
+	}
+	return mirrorSet{source: extractHostname(srcImage), mirrors: truncatedMirrors, mirrorSourcePolicy: mirrorSourcePolicy}
+}
+
+// extractHostname extracts just the initial host repo from a full image location
+// e.g. mcr.microsoft.com would be extracted from mcr.microsoft.com/oss/kubernetes/pause:3.9
+func extractHostname(fullImage string) string {
+	parts := strings.Split(fullImage, imagePathSeparator)
+	return parts[0]
 }
 
-// NewRegistryConfig creates a new RegistryConfig object by extracting and merging the contents of the given mirror sets
-func NewRegistryConfig(idmsItems []config.ImageDigestMirrorSet, idtsItems []config.ImageTagMirrorSet) *registryConfig {
+// getMirrorSets extracts and merges the contents of the given mirror sets.
```

Comment:
This function is getting and merging the mirror sets consider either renaming it to denote that or doing the merge where this is being called.

Reply:
sure, updated the name